### PR TITLE
Support InfiniAI Megrez 3b

### PR DESCRIPTION
**Pull request by dixyes**
This pr is to add [InfiniAI Megrez](https://huggingface.co/Infinigence/Megrez-3B-Instruct) support into llama.cpp

The model now(@58f1df16523cb2a9acb225aa808146e052f2b5b2) seems have a wrong `eos_token` set in [its tokenizer_config.json](https://huggingface.co/Infinigence/Megrez-3B-Instruct/blob/58f1df16523cb2a9acb225aa808146e052f2b5b2/tokenizer_config.json#L214).
( `<|turn_end|>` at template and `<|turn_end>` in json) Not sure if this is on purpose. Also metioned [here](https://github.com/infinigence/Infini-Megrez/issues/2#issuecomment-2553108827)

So the converted model will not stop generating in chat mode. Modify it to `<|turn_end|>` in `tokenizer_config.json`, then the generated gguf will work.

### CODE REVIEW

**Review by ngxson — COMMENTED** [review]
I'm not sure about the `tokenizer_pre == "megrez"` part (if other collaborators know, please feel free to review this PR).

The template part looks good to me.

File: tests/test-chat-template.cpp [hunk 1]
```diff
@@ -77,6 +77,8 @@ int main(void) {
...
+        // Infinigence/Megrez-3B-Instruct
+        u8"{% for message in messages %}{% if loop.first and messages[0]['role'] != 'system' %}{{ '<|role_start|>system<|role_end|>你是Megrez-3B-Instruct，将针对用户的问题给出详细的、积极的回答。<|turn_end|>' }}{% endif %}{{ '<|role_start|>' + message['role'] + '<|role_end|>' + message['content'] + '<|turn_end|>' }}{% endfor %}{% if add_generation_prompt %}{{ '<|role_start|>assistant<|role_end|>' }}{% endif %}"
```
Comment (arch-btw) [inline tests/test-chat-template.cpp:81]: is the u8 supposed to be here?

**Comment by arch-btw** [comment]
Thanks for doing this, I was trying it myself but didn't finish it. Just so you know they fixed the eos 30 minutes ago.

File: src/llama.cpp [hunk 2]
```diff
@@ -6602,7 +6604,8 @@ static void llm_load_vocab(
                 vocab.type_pre = LLAMA_VOCAB_PRE_TYPE_COMMAND_R;
                 vocab.tokenizer_clean_spaces = false;
             } else if (
-                tokenizer_pre == "qwen2") {
+                tokenizer_pre == "qwen2" ||
+                tokenizer_pre == "megrez") {
```
Comment (slaren) [inline src/llama.cpp:6608]: This sets `vocab.tokenizer_clean_spaces` to `false`, but 
https://huggingface.co/Infinigence/Megrez-3B-Instruct/blob/main/tokenizer_config.json has `"clean_up_tokenization_spaces": true,`. Is this correct?

Comment (dixyes) [reply src/llama.cpp]: My bad, not noticed this. Fixed using another if branch.

**Review by slaren — APPROVED** [review]

**Pull request merged**